src/prepared_discover/prepared_discover_txn.c

```diff
@@ -66,23 +102,24 @@ int
 __wti_prepared_discover_add_artifact_upd(
   WT_SESSION_IMPL *session, wt_timestamp_t prepare_transaction_id, WT_ITEM *key, WT_UPDATE *upd)
 {
-    WT_DECL_RET;
-    WT_SESSION_IMPL *prep_session;
+    WT_PENDING_PREPARED_ITEM *prepared_item;
+    WT_TXN_OP *op;
 
-    WT_RET(__wt_prepared_discover_find_or_create_transaction(
-      session, prepare_transaction_id, &prep_session, true));
+    WT_RET(
+      __prepared_discover_find_or_create_item(session, prepare_transaction_id, &prepared_item));
 
-    /* Add the update to the prepared transaction context. */
-    WT_WITH_DHANDLE(prep_session, session->dhandle, ret = __wt_txn_modify(prep_session, upd));
-    WT_RET(ret);
-    /* Copy the key into the transaction operation, so it can be used to find the update later. */
-    WT_WITH_DHANDLE(prep_session, session->dhandle, ret = __wt_txn_op_set_key(prep_session, key));
-    WT_RET(ret);
+    WT_RET(__wt_pending_prepared_next_op(session, &op, prepared_item));
+    WT_RET(__wt_op_modify(session, upd, op));
 
+    WT_ASSERT(session, op->type == WT_TXN_OP_BASIC_ROW || op->type == WT_TXN_OP_INMEM_ROW);
+    /*
+     * Copy the key into the transaction operation structure, so when update is evicted to the
+     * history store, we can still find it.
+     */
```

Comment:
Can we move this code to __wt_pending_prepared_next_op? Better to put them together for understanding?

Reply:
done